### DISCUSSION TITLE
Fast API @get is responding to OPTIONS requests sent by browser CORS negotiation (Not handled by middleware)

### DISCUSSION THREAD

**Discussion by blipk** [fastapi/fastapi#9564] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.responses import JSONResponse
from starlette.middleware.cors import CORSMiddleware


app = FastAPI(
    title="MinimalCorsExample",
    default_response_class=JSONResponse,
)


API_CORS_ORIGINS = ["http://localhost", "https://localhost", "https://localhost:1234", "https://localhost:8000", "https://localhost:5173", "http://localhost:5173"]
app.add_middleware(
    CORSMiddleware,
    allow_origins=[str(origin) for origin in API_CORS_ORIGINS],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

router = APIRouter(
    prefix=f"/dashboard",
    tags=["Dashboard"],
)


app.include_router(router)

@router.get("/")
async def get_dashboard(*, request: Request, response: Response):
    print(request.method)
    if request.method == "OPTIONS":
    	return {"response": "Empty response to avoid database query"}
    	
    # Database query here...
    db_data = []
    
    return {"response": db_data}             

if __name__ == "__main__":
    uvicorn.run("main:app")
```

### Description

Sometimes when browsers make a fetch request (specifically with AXIOS in my case) they will often send an OPTIONS request with the same URL and body to receive the CORS headers before making the GET request.

Consider the code I've provided, the `@router.get("/")` on `get_dashboard()` is still being sent OPTIONS requests to be handled, thus duplicating the request when the following GET request comes through. 

(This is why I check request.method and send an empty response for the initial OPTIONS request)

This should be handled in the middleware by sending an empty/appropriate response with the CORS headers on the OPTIONS request.
And I would assume `.get()` would only be for actual GET requests, but it's not the case.

Please let me know if I'm doing something wrong or this is expected behaviour.

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

Python 3.10.10

**Comment by JavierSanchezCastro** [comment]
You can use @router.options as defined here:

[https://fastapi.tiangolo.com/tutorial/first-steps/#define-a-path-operation-decorator](url)
[https://fastapi.tiangolo.com/tutorial/cors/#steps](url)

Also, check this line, in your code example should be at the end
```python3
app.include_router(router)
```

Example:
```python3
from fastapi.responses import JSONResponse, Response
from fastapi import FastAPI, APIRouter, Request
from starlette.middleware.cors import CORSMiddleware


app = FastAPI(
    title="MinimalCorsExample",
    default_response_class=JSONResponse,
)

API_CORS_ORIGINS = ["http://localhost", "https://localhost", "https://localhost:1234", "https://localhost:8000", "https://localhost:5173", "http://localhost:5173"]
app.add_middleware(
    CORSMiddleware,
    allow_origins=[API_CORS_ORIGINS],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

router = APIRouter(
    prefix=f"/dashboard",
    tags=["Dashboard"],
)

@router.get("/")
async def get_dashboard(*, request: Request, response: Response):
    db_data = []
    
    return {"response": db_data}

@router.options("/")
async def options_root():
    allowed_methods = ["GET", "OPTIONS"]
    headers = {
        "Allow": ", ".join(allowed_methods)
    }
    return JSONResponse(content=None, headers=headers)

app.include_router(router)
```
